src/Models/EvaluationResult.php

```
@@ -80,6 +80,20 @@ public static function findById(Database $db, int $id): ?array
         return $row !== false ? $row : null;
     }
 
+    /**
+     * Find a single evaluation result with a FOR UPDATE lock (use inside a transaction).
+     *
+     * @param Database $db Database instance
+     * @param int      $id Result primary key
+     * @return array|null  Row as associative array, or null if not found
+     */
+    public static function findByIdForUpdate(Database $db, int $id): ?array
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $db. Configured minimum length is 3. (undefined)

(ShortVariable)